db/sqldb.go

```diff
@@ -0,0 +1,557 @@
+
+	// Itterate over all of the rows creating the slice of volumes we need. (there should only ever be one or zero)
+	var vols []*volume.Volume
+	var ids []int
+	for rows.Next() {
+		var id int
+		var name string
+		var mountpointNS sql.NullString
+		err = rows.Scan(&id, &name, &mountpointNS)
+		if err != nil {
+			return nil, 0, err
+		}
+
+		// If there is a mountpoint, use it.
+		var mountpoint string
+		if mountpointNS.Valid {
+			mountpoint = mountpointNS.String
+		}
+
+		// Create volume
+		vol := volume.Volume{
+			Name:       name,
+			Mountpoint: mountpoint,
+			Status:     nil}
+
+		// Append it to the list
+		vols = append(vols, &vol)
+		ids = append(ids, id)
+	}
+
+	// Check to see if there was an error durring interation
+	err = rows.Err()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// Did we get any results?
+	if len(vols) == 0 {
+		return nil, 0, errors.New("Volume does not exist.")
+	}
+
+	return vols[0], ids[0], nil
+}
+
+// Path returns the mountpath of a particular volume
+func (s SQLVolumeDatabase) Path(volumeName string) (string, error) {
+	vol, err := s.Get(volumeName)
+	if err != nil {
+		return "", err
+	}
+
+	return vol.Mountpoint, nil
+}
+
+// Remove (Delete) a volume from the database.
+func (s SQLVolumeDatabase) Remove(volumeName string) error {
+	_, requests, err := s.listMounts(volumeName)
+	if err != nil {
+		return err
+	}
+
+	if requests > 0 {
+		return errors.New("Volume cannot be removed as it still has active mount requests.")
+	}
+
+	id, err := s.getVolumeIDByName(volumeName)
+	if err != nil {
+		return err
+	}
+
+	// Begin transaction to the database
+	tx, err := sqlDB.Begin()
+	if err != nil {
+		return err
+	}
+
+	// Prepare the queries
+	stmtM, err := tx.Prepare(s.DBQueries.mountsDeleteByVolumeIDSQL)
+	if err != nil {
+		return err
+	}
+	defer stmtM.Close()
+
+	stmtV, err := tx.Prepare(s.DBQueries.volumesDeleteByIDSQL)
+	if err != nil {
+		return err
+	}
+	defer stmtV.Close()
+
+	// Actually make the deletes
+	_, err = stmtM.Exec(id)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	_, err = stmtV.Exec(id)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	// Commit the change
+	return tx.Commit()
+}
+
+// Mount the volume with name and id
+func (s SQLVolumeDatabase) Mount(volumeName string, id string) (string, error) {
+	mounts, _, err := s.listMounts(volumeName)
+	if err != nil {
+		return "", err
+	}
+
+	vol, volid, err := s.getVolumeByName(volumeName)
+	if err != nil {
+		return "", err
+	}
+
+	// Begin transaction to the database
+	tx, err := sqlDB.Begin()
+	if err != nil {
+		return "", err
+	}
+
+	if vol.Mountpoint == "" {
+		vol.Mountpoint = "/etc/docker/mounts/" + volumeName
+		stmtUp, errUp := tx.Prepare(s.DBQueries.volumesUpdateMountpointSQL)
+		if errUp != nil {
+			return "", errUp
+		}
+		defer stmtUp.Close()
+
+		_, errUp = stmtUp.Exec(vol.Mountpoint, volid)
+		if errUp != nil {
+			tx.Rollback()
+			return "", errUp
+		}
+	}
+
+	var newCount int
+	var q string
```

Comment:
Short variable names hinder the reader's ability to understand this code. Please lengthen them for all our sakes.